src/circuit.rs

```diff
@@ -215,10 +828,33 @@ mod tests {
                 let value = spent_note.value() - output_note.value();
                 let cv_net = ValueCommitment::derive(value.unwrap(), ValueCommitTrapdoor::zero());
 
+                let path = MerklePath::dummy(&mut rng);
+                let anchor = path.root(spent_note.commitment().into()).unwrap();
+
                 (
-                    Circuit {},
+                    Circuit {
+                        path: Some(path.auth_path()),
+                        pos: Some(path.position()),
+                        g_d_old: Some(sender_address.g_d()),
+                        pk_d_old: Some(*sender_address.pk_d()),
+                        v_old: Some(spent_note.value()),
+                        rho_old: Some(spent_note.rho()),
+                        psi_old: Some(spent_note.rseed().psi(&spent_note.rho())),
+                        rcm_old: Some(spent_note.rseed().rcm(&spent_note.rho())),
+                        cm_old: Some(spent_note.commitment()),
+                        alpha: Some(alpha),
+                        ak: Some(ak),
+                        nk: Some(nk),
+                        rivk: Some(rivk),
+                        g_d_new_star: Some((*output_note.recipient().g_d()).to_bytes()),
+                        pk_d_new_star: Some(output_note.recipient().pk_d().to_bytes()),
+                        v_new: Some(output_note.value()),
+                        psi_new: Some(output_note.rseed().psi(&output_note.rho())),
+                        rcm_new: Some(output_note.rseed().rcm(&output_note.rho())),
+                        rcv: Some(ValueCommitTrapdoor::zero()),
+                    },
                     Instance {
-                        anchor: pallas::Base::zero().into(),
+                        anchor,
                         cv_net,
                         nf_old,
                         rk,
```

Comment:
This always passes `enable_spend: true`; I gather that means it does not support building coinbase transactions?

Reply:
The round_trip test does pass `enable_spend: true` now, but we should test `enable_spend: false` too.